src/lib/Menu/MainMenuBuilder.php

```diff
@@ -173,95 +174,85 @@ public function createStructure(array $options): ItemInterface
         $token = $this->tokenStorage->getToken();
 
         /** @var \Knp\Menu\ItemInterface|\Knp\Menu\ItemInterface[] $menu */
-        $menu = $this->factory->createItem('root');
+        $menu = $this->createMenuItem('root');
 
-        $menu->addChild($this->factory->createItem(self::ITEM_DASHBOARD, [
-                'route' => 'ezplatform.dashboard',
-                'attributes' => [
-                    'data-tooltip-placement' => 'right',
-                    'data-tooltip-extra-class' => 'ibexa-tooltip--info-neon',
-                ],
-                'extras' => [
-                    'icon' => 'dashboard-clean',
-                    'orderNumber' => 20,
-                ],
-            ]
-        ));
+        $menu->addChild($contentMenu = $this->createMenuItem(self::ITEM_CONTENT, [
+            'attributes' => [
+                'data-tooltip-placement' => 'right',
+                'data-tooltip-extra-class' => 'ibexa-tooltip--info-neon',
+            ],
+            'extras' => [
+                'icon' => 'hierarchy',
+                'orderNumber' => 40,
+            ],
+        ]));
 
-        $menu->addChild($this->factory->createItem(self::ITEM_CONTENT, [
-                'attributes' => [
-                    'data-tooltip-placement' => 'right',
-                    'data-tooltip-extra-class' => 'ibexa-tooltip--info-neon',
-                ],
-                'extras' => [
-                    'icon' => 'hierarchy',
-                    'orderNumber' => 40,
-                ],
-            ]
-        ));
+        $contentMenu->addChild($this->createMenuItem(self::ITEM_DASHBOARD, [
+            'route' => 'ezplatform.dashboard',
+            'attributes' => [
+                'data-tooltip-placement' => 'right',
+                'data-tooltip-extra-class' => 'ibexa-tooltip--info-neon',
+            ],
+            'extras' => [
+                'icon' => 'dashboard-clean',
+                'orderNumber' => 20,
+            ],
+        ]));
 
-        $menu->addChild($this->factory->createItem(self::ITEM_ADMIN, [
-                'attributes' => [
-                    'data-tooltip-placement' => 'right',
-                    'data-tooltip-extra-class' => 'ibexa-tooltip--info-neon',
-                ],
-                'extras' => [
-                    'separate' => true,
-                    'bottom_item' => true,
-                    'icon' => 'settings-block',
-                    'orderNumber' => 140,
-                ],
-            ]
-        ));
+        $menu->addChild($adminMenu = $this->createMenuItem(self::ITEM_ADMIN, [
```

Comment:
`ItemInterface::addChild()` returns the added child. You can assign from the returned value, instead of inlining the assignment (which is usually frowned upon, because it's not "immediately" visible)

```suggestion
        $adminMenu = $menu->addChild($this->createMenuItem(self::ITEM_ADMIN, [
```